File: mvp/sdk/python/saas_ia/api/pdf.py
"""PDF Processor API — text/table extraction (PyMuPDF + pdfplumber)."""

from __future__ import annotations

from typing import Any, BinaryIO

from saas_ia.api.base import BaseAPI


class PDFAPI(BaseAPI):
    """Wraps ``/api/pdf`` endpoints."""

    async def process(
        self,
        file: BinaryIO,
        filename: str = "document.pdf",
    ) -> dict[str, Any]:
        """Upload and process a PDF file."""
        return await self._post(
            "/api/pdf/process",
            files={"file": (filename, file)},
        )

    async def list(self) -> list[dict[str, Any]]:
        """List processed PDFs."""
        return await self._get("/api/pdf/")

    async def get(self, pdf_id: str) -> dict[str, Any]:
        """Get a processed PDF result."""
        return await self._get(f"/api/pdf/{pdf_id}")

    async def delete(self, pdf_id: str) -> None:
        """Delete a processed PDF."""
        await self._delete(f"/api/pdf/{pdf_id}")

File: mvp/sdk/typescript/src/api/pdf.ts
import { BaseAPI } from "./base";
import { PdfResult } from "../types";

/**
 * PDF Processor API — extract text and tables from PDF files using
 * PyMuPDF + pdfplumber.
 */
export class PDFAPI extends BaseAPI {
  /** Upload and process a PDF file. */
  async process(file: Blob | File): Promise<PdfResult> {
    const form = new FormData();
    form.append("file", file);
    return this._request<PdfResult>("/api/pdf/process", {
      method: "POST",
      body: form,
    });
  }

  /** List processed PDFs. */
  async list(): Promise<PdfResult[]> {
    return this._get<PdfResult[]>("/api/pdf/");
  }

  /** Get a processed PDF result. */
  async get(id: string): Promise<PdfResult> {
    return this._get<PdfResult>(`/api/pdf/${id}`);
  }

  /** Delete a processed PDF. */
  async delete(id: string): Promise<void> {
    return this._delete(`/api/pdf/${id}`);
  }
}
